src/main/java/Duke.java

```diff
@@ -1,10 +1,183 @@
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String HORIZONTAL_LINE =
+            "____________________________________________________________";
+
+    private static final int REQUIRED_NUMBER_OF_ARGUMENT_FOR_TODO = 2;
+    private static final int REQUIRED_NUMBER_OF_ARGUMENT_FOR_EVENT = 4;
+    private static final int REQUIRED_NUMBER_OF_ARGUMENT_FOR_DEADLINE = 4;
+
+    private static void greetUser() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println(" Hello! I'm Duke");
+        System.out.println(" What can I do for you?");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static String[] parseUserInput(String userInput) {
+        ArrayList<String> inputArguments = new ArrayList<>();
+        String[] splitInput = userInput.split(" ");
+        String userCommand = splitInput[0];
+        inputArguments.add(userCommand);
+
+        // Build arguments (if any): description and command option (at/by) information (only 1 is expected)
+        String commandOption = "";
+        StringBuilder sbArgument = new StringBuilder();
+        for (int i = 1; i < splitInput.length; i++) {
+            boolean isFirstCommandOption = commandOption.isEmpty() && splitInput[i].startsWith("/");
+            if (isFirstCommandOption) {
+                String description = sbArgument.toString().stripTrailing();
+                commandOption = splitInput[i];
+                inputArguments.add(description);
+                inputArguments.add(commandOption);
+                sbArgument.setLength(0);
+                continue;
+            }
+
+            sbArgument.append(splitInput[i]).append(" ");
+        }
+
+        // Depending on user input, it could be description or task index or at/by information or empty
+        String lastArgument = sbArgument.toString().stripTrailing();
+        inputArguments.add(lastArgument);
+        return inputArguments.toArray(new String[0]);
+    }
+
+    private static void addTask(ArrayList<Task> taskList, Task newTask) {
+        pendingTaskCount++;
+        taskList.add(newTask);
+        System.out.println(" Gotcha! I've added this task: ");
+        System.out.println("\t" + newTask.toString());
+        System.out.println(" There's currently " + taskList.size() + " task(s) in the list." );
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        greetUser();
+
+        boolean shouldExit = false;
+        ArrayList<Task> taskList = new ArrayList<>();
```

Comment:
Perhaps can declare the maximum size of the array list?

Reply:
I think it may not be necessary since no maximum size is specified for this iP.